File: frontend/src/services/sessionService.ts
import api, { ApiResponse } from './api';
import { BrainstormingSession, SessionDetail, CreateSessionRequest, Round } from '../types';

export const sessionService = {
  async getByTeamId(teamId: string): Promise<BrainstormingSession[]> {
    const response = await api.get<ApiResponse<BrainstormingSession[]>>(`/sessions/team/${teamId}`);
    return response.data.data || [];
  },

  async getByTopicId(topicId: string): Promise<BrainstormingSession[]> {
    const response = await api.get<ApiResponse<BrainstormingSession[]>>(`/sessions/topic/${topicId}`);
    return response.data.data || [];
  },

  async getById(id: string): Promise<SessionDetail> {
    const response = await api.get<ApiResponse<SessionDetail>>(`/sessions/${id}`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Session not found');
    }
    return response.data.data;
  },

  async create(data: CreateSessionRequest): Promise<BrainstormingSession> {
    const response = await api.post<ApiResponse<BrainstormingSession>>('/sessions', data);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to create session');
    }
    return response.data.data;
  },

  async start(id: string): Promise<BrainstormingSession> {
    const response = await api.post<ApiResponse<BrainstormingSession>>(`/sessions/${id}/start`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to start session');
    }
    return response.data.data;
  },

  async pause(id: string): Promise<BrainstormingSession> {
    const response = await api.post<ApiResponse<BrainstormingSession>>(`/sessions/${id}/pause`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to pause session');
    }
    return response.data.data;
  },

  async resume(id: string): Promise<BrainstormingSession> {
    const response = await api.post<ApiResponse<BrainstormingSession>>(`/sessions/${id}/resume`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to resume session');
    }
    return response.data.data;
  },

  async end(id: string): Promise<BrainstormingSession> {
    const response = await api.post<ApiResponse<BrainstormingSession>>(`/sessions/${id}/end`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to end session');
    }
    return response.data.data;
  },

  async advanceRound(id: string): Promise<Round> {
    const response = await api.post<ApiResponse<Round>>(`/sessions/${id}/advance-round`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to advance round');
    }
    return response.data.data;
  },

  async getCurrentRound(id: string): Promise<Round> {
    const response = await api.get<ApiResponse<Round>>(`/sessions/${id}/current-round`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Round not found');
    }
    return response.data.data;
  },

  async getRemainingTime(id: string): Promise<{ remainingSeconds: number }> {
    const response = await api.get(`/sessions/${id}/remaining-time`);
    return response.data;
  },

  async getStatus(id: string): Promise<{ status: string }> {
    const response = await api.get(`/sessions/${id}/status`);
    return response.data;
  },
};

File: mobile/lib/services/session_service.dart
import '../core/constants/api_constants.dart';
import '../models/models.dart';
import 'api_client.dart';

class SessionService {
  final ApiClient _apiClient = ApiClient();

  Future<List<BrainstormingSession>> getSessions({String? teamId}) async {
    final queryParams = teamId != null ? {'teamId': teamId} : null;
    final response = await _apiClient.get(
      ApiConstants.sessions,
      queryParameters: queryParams,
    );
    final List<dynamic> data = response.data;
    return data.map((json) => BrainstormingSession.fromJson(json)).toList();
  }

  Future<SessionDetail> getSession(String id) async {
    final response = await _apiClient.get('${ApiConstants.sessions}/$id');
    return SessionDetail.fromJson(response.data);
  }

  Future<BrainstormingSession> createSession(CreateSessionRequest request) async {
    final response = await _apiClient.post(
      ApiConstants.sessions,
      data: request.toJson(),
    );
    return BrainstormingSession.fromJson(response.data);
  }

  Future<BrainstormingSession> startSession(String id) async {
    final response = await _apiClient.post('${ApiConstants.sessions}/$id/start');
    return BrainstormingSession.fromJson(response.data);
  }

  Future<BrainstormingSession> pauseSession(String id) async {
    final response = await _apiClient.post('${ApiConstants.sessions}/$id/pause');
    return BrainstormingSession.fromJson(response.data);
  }

  Future<BrainstormingSession> resumeSession(String id) async {
    final response = await _apiClient.post('${ApiConstants.sessions}/$id/resume');
    return BrainstormingSession.fromJson(response.data);
  }

  Future<BrainstormingSession> endSession(String id) async {
    final response = await _apiClient.post('${ApiConstants.sessions}/$id/end');
    return BrainstormingSession.fromJson(response.data);
  }

  Future<Round> advanceRound(String id) async {
    final response = await _apiClient.post('${ApiConstants.sessions}/$id/advance-round');
    return Round.fromJson(response.data);
  }

  Future<void> deleteSession(String id) async {
    await _apiClient.delete('${ApiConstants.sessions}/$id');
  }

  // Get sessions for current user
  Future<List<BrainstormingSession>> getMySessions() async {
    final response = await _apiClient.get('${ApiConstants.sessions}/my');
    final List<dynamic> data = response.data;
    return data.map((json) => BrainstormingSession.fromJson(json)).toList();
  }

  // Get sessions for a team
  Future<List<BrainstormingSession>> getSessionsByTeam(String teamId) async {
    final response = await _apiClient.get('${ApiConstants.sessions}/team/$teamId');
    final List<dynamic> data = response.data;
    return data.map((json) => BrainstormingSession.fromJson(json)).toList();
  }

  // Get current round info
  Future<Round?> getCurrentRound(String sessionId) async {
    try {
      final response = await _apiClient.get('${ApiConstants.sessions}/$sessionId/current-round');
      return Round.fromJson(response.data);
    } catch (e) {
      return null;
    }
  }
}
